plugins/modules/foreman_role.py

```diff
@@ -101,7 +118,20 @@ def main():
         if 'organizations' in entity_dict:
             entity_dict['organizations'] = module.find_resources_by_name('organizations', entity_dict['organizations'], thin=True)
 
-    changed = module.ensure_entity_state('roles', entity_dict, entity)
+    filters = entity_dict.pop("filters", None)
+
+    changed, entity = module.ensure_entity('roles', entity_dict, entity)
+
+    if not module.desired_absent and not module.check_mode:
+        if filters is not None:
+            for role_filter in filters:
+                role_filter['role_id'] = entity['id']
+                role_filter['permissions'] = module.find_resources_by_name('permissions', role_filter['permissions'])
```

Comment:
correct, but given the `permissions` endpoint doesn't really support this, the performance will be identical.

@Fodoj I'd prefer if you could fix that for consistency reasons, but I wouldn't block the PR on that. Please tell me if you have time, otherwise I'd merge as-is and follow up with a cleanup later.

Reply:
Done!